webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/PreferencesUtil.java

```diff
@@ -114,6 +125,14 @@ public static void loadPreferences(String aUsername, SettingsService aSettingsSe
                         .getProject());
                 aBModel.setAnnotationLayers(layers);
             }
+
+            // Get color preferences for each layer, init with legacy if not found
```

Comment:
Did you try what happens if `(preference.getColorPerLayer() != null` and an admin adds a new layer? Will there be an exception? What coloring strategy will it have?